### PR TITLE
Update to libswiftnav.

### PR DESCRIPTION
**Pull request by mookerji**
Includes:
- Travis integration.
- libswiftnav submodule
- Python setup tools update.

/cc @fnoble @henryhallam @benjamin0 @cbeighley @peddie

### CODE REVIEW

**Comment by mookerji** [comment]
WIP! Testing CI stuff.

**Comment by mookerji** [comment]
Don't quite know why travis is failing here. There's currently a Vagrant installer which gets everything setup and working on Linux, so feel free to branch off to update the tracking/navigation stuff. 

If there's an example of some working code invoking peregrine on some test data I could also use that to update it in a separate PR.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+sudo: false
+language: python
+cache:
+  pip: true
```
Comment (benjamin0) [inline .travis.yml]: this wasn't working for me on sbp_log_analysis so I used 
```
cache:
  directories:
    - $HOME/.cache/pip
```

File: deps.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+         python-cython \
+         python-dev
+    git submodule update --init
+    # Build libswiftnav
+    cd libswiftnav/
+    mkdir -p build && cd build/
+    cmake ../
+    make
+    # Install locally to avoid sudo
+    sudo make install
```
Comment (cbeighley) [inline deps.sh]: Comment says install locally but it's installing globally?

File: deps.sh [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+function install_deps_osx () {
+    # TODO: Add OS X brew installation dependencies
+    git submodule update --init
+    # Build libswiftnav
+    cd libswiftnav/
+    mkdir -p build && cd build/
+    cmake ../
+    make
+    # Install locally to avoid sudo
```
Comment (cbeighley) [inline deps.sh]: Same

Comment (mookerji) [inline deps.sh]: Done

File: deps.sh (hunk 2 above)
Comment (mookerji) [inline deps.sh]: Done

File: .travis.yml (hunk 1 above)
Comment (mookerji) [inline .travis.yml]: Done

**Pull request merged**